### PR TITLE
ex 17 implemented

### PR DESCRIPTION
**Pull request by omiras**

### CODE REVIEW

File: public/js/translations.js [hunk 1]
```diff
@@ -222,6 +222,15 @@ var resources = {
         'instructions': `<div>Our bugs are starving, use the <strong>map()</strong> methods to double their weight. You must assign the new array into <strong>feededBugs</strong> <br><br>Each bug has 2 properties: name and weight.<br>--> [{name:'ladybug', weight: '10'}]</div>`,
         'context': `` // levels.js -> contextInstructions
       },
+      'level_17':
+      {
+        'methodTitle': `Crate a new array with the bugs which name has more than 5 letters`,
```
Comment (Andreu-G) [inline public/js/translations.js:227]:
```suggestion
        'methodTitle': `Create a new array with the name of the bugs that have more than 5 letters`,
```

File: public/js/translations.js [hunk 2]
```diff
@@ -222,6 +222,15 @@ var resources = {
...
+        'methodHelp': `Filter the bugs`,
+        'methodName': `Filter method`,
+        'desc': `The filter() method creates a new array with the elements that pass the test implemented by the provided function.`,
+        'instructions': `<div>We like large bugs names, so in this exercise, use the <strong>filter</strong> method to achieve it</div>`,
```
Comment (Andreu-G) [inline public/js/translations.js:231]:
```suggestion
        'instructions': `<div>We like long bugs names, so in this exercise, use the <strong>filter()</strong> method to achieve it</div>`,
```

File: public/js/translations.js [hunk 3]
```diff
@@ -431,6 +440,15 @@ var resources = {
         'instructions': `<div>Nuestros insectos están hambrientos, usa el <i>método</i> <strong>map()</strong> para duplicar su peso. Debes asignar el nuevo array en la variable <strong>feededBugs</strong></div>`,
         'context': `` // levels.js -> contextInstructions
       },
+      'level_17':
+      {
+        'methodTitle': `Crea un nuevo array con el nombre de los bichos que tienen más de 5 letras`,
```
Comment (Andreu-G) [inline public/js/translations.js:445]:
```suggestion
        'methodTitle': `Crea un nuevo array con el nombre de los insectos que tienen más de 5 letras`,
```

File: public/js/translations.js [hunk 4]
```diff
@@ -431,6 +440,15 @@ var resources = {
...
+        'methodHelp': `Filtra los bichos`,
```
Comment (Andreu-G) [inline public/js/translations.js:446]:
```suggestion
        'methodHelp': `Filtra los insectos`,
```

File: public/js/translations.js [hunk 5]
```diff
@@ -431,6 +440,15 @@ var resources = {
...
+        'methodName': `Método <em>filter</em>`,
+        'desc': `El método filter crea un nuevo array con todos elementos que pasan la función de test.`,
```
Comment (Andreu-G) [inline public/js/translations.js:448]:
```suggestion
        'desc': `El método filter() crea un nuevo array con todos elementos que pasan la función de test.`,
```

File: public/js/translations.js [hunk 6]
```diff
@@ -431,6 +440,15 @@ var resources = {
...
+        'instructions': `<div>Nos gustan los animales con nombres largos. Usa el método <strong>filter</strong> para pasar este ejercicio.</div>`,
```
Comment (Andreu-G) [inline public/js/translations.js:449]:
```suggestion
        'instructions': `<div>Nos gustan los insectos con nombres largos. Usa el método <strong>filter()</strong> para pasar este ejercicio.</div>`,
```

File: public/js/levels.js [hunk 7]
```diff
@@ -1069,6 +1069,58 @@ var levels = [
...
+      </bracket>
+    `,
+    boardMarkupSolution: `
+      <bracket>
+        <grass>
+          <butterfly class="dance">
+        </grass>
+        <grass>
+          <dragonFly class="dance">
+        </grass>
```
Comment (Andreu-G) [inline public/js/levels.js:1118]:
```suggestion
        </grass>
        <grass>
          <spider class="dance">
        </grass>
```

File: public/js/levels.js [hunk 8]
```diff
@@ -1069,6 +1069,58 @@ var levels = [
...
+          <butterfly class="dance">
+        </grass>
+        <grass>
+          <bee>
+        </grass>
+        <grass>
+          <dragonFly class="dance">
+        </grass>
+        <grass>
+          <spider>
```
Comment (Andreu-G) [inline public/js/levels.js:1107]:
```suggestion
          <spider class="dance">
```

File: public/js/levels.js [hunk 9]
```diff
@@ -1069,6 +1069,58 @@ var levels = [
...
+      help: 'The filter() method create a new array with the elements that pass the test implemented by the provided function.',
+      examples: [
+        `<strong>1</strong>const words = ['spray', 'limit', 'elite', 'exuberant', 'destruction', 'present'];
+        <br><strong>2</strong>const result = words.filter(word => word.startsWith("e"));
+        ;<br> <strong>3</strong> console.log(result);        `,
+        '<strong>TERMINAL</strong> <br> <div class="console-wrapper"> ["elite", "exuberant"]</div>',
+      ],
+      myGrass : "var myGrass = ['worm','butterfly','bee','dragonFly','spider'];\n\nvar largeNames;",
+      variableToCheck: "myGrass",
+      myGrassSolution : ['butterfly','dragonFly'],
```
Comment (Andreu-G) [inline public/js/levels.js:1089]:
```suggestion
      myGrassSolution : ['butterfly','dragonFly','spider'],
```
Spider has 6 letters :)

File: public/js/levels.js [hunk 10]
```diff
@@ -1069,6 +1069,58 @@ var levels = [
     `,
     instructions: "<div>Our bugs are starving, use the <strong>map()</strong> methods to double their weight. You must assign the new array into <strong>feededBugs</strong> <br><br>Each bug has 2 properties: name and weight.<br>--> [{name:'ladybug', weight: '10'}]</div>" 
     },
+    // level 17 : FILTER
+    {
+      helpTitle: "Creates a copy of a portion of the original array",
+      selectorName: "Filter() method",
+      doThis: "Crate a new array with the bugs which name has more than 5 letters",
```
Comment (Andreu-G) [inline public/js/levels.js:1076]:
```suggestion
      doThis: "Create a new array with the name of the bugs that have more than 5 letters",
```

**Review by Andreu-G — CHANGES_REQUESTED** [review]
Other than the comments above, I also believe the *largeNames* variable is currently useless.

Screenshot_000750

As you can see, the code does not complete. I believe this is because you're not actually changing the *myGrass* variable, but instead the *largeNames* variable, which the code doesn't care about.

Screenshot_000751

If you instead modify *myGrass* directly, the level is completed.

File: public/js/levels.js [hunk 11]
```diff
@@ -1069,6 +1069,58 @@ var levels = [
...
+      syntax: "filter();",
+      regExp: "myGrass\.filter\(\);?",
+      help: 'The filter() method create a new array with the elements that pass the test implemented by the provided function.',
+      examples: [
+        `<strong>1</strong>const words = ['spray', 'limit', 'elite', 'exuberant', 'destruction', 'present'];
+        <br><strong>2</strong>const result = words.filter(word => word.startsWith("e"));
+        ;<br> <strong>3</strong> console.log(result);        `,
+        '<strong>TERMINAL</strong> <br> <div class="console-wrapper"> ["elite", "exuberant"]</div>',
+      ],
+      myGrass : "var myGrass = ['worm','butterfly','bee','dragonFly','spider'];\n\nvar largeNames;",
```
Comment (Andreu-G) [inline public/js/levels.js:1087]:
```suggestion
      myGrass : "var myGrass = ['worm','butterfly','bee','dragonFly','spider'];\n\nvar longNames;",
```

File: public/js/levels.js [hunk 12]
```diff
@@ -1069,6 +1069,58 @@ var levels = [
...
+      variableToCheck: "myGrass",
```
Comment (Andreu-G) [inline public/js/levels.js:1088]:
```suggestion
      variableToCheck: "longNames",
```

**Comment by Andreu-G** [comment]
With these 2 last changes, everything should work perfectly.

**Comment by omiras** [comment]
OMG thank you @Andreu-G , I think that I even did not test it, I was too confident!

**Pull request merged**